File: php/Memory.php
<?php
include("Hardware.php");

class Memory extends Hardware
{
    private $frequency = 0;
    private $memorySize = 0;
    private $supportCuda = false;

    public function __construct($frequency = 0, $memorySize = 0, $supportCuda = false)
    {
        $this->frequency = $frequency;
        $this->memorySize = $memorySize;
        $this->supportCuda = $supportCuda;
    }

    public function getFrequency()
    {
        return $this->frequency;
    }
    public function setFrequency($frequency)
    {
        $this->frequency = $frequency;
    }

    public function getMemorySize()
    {
        return $this->memorySize;
    }
    public function setMemorySize($memorySize)
    {
        $this->memorySize = $memorySize;
    }

    public function isSupporCuda()
    {
        return $this->supportCuda;
    }
    public function setSupportCuda($supportCuda)
    {
        $this->supportCuda = $supportCuda;
    }

    public function printMemory()
    {
        echo "Frequency: " . $this->getFrequency() . "Mhz<br/>";
        echo "Memory Size: " . $this->getMemorySize() . "GB<br/>";
        echo "Support Cuda: ";
        if ($this->isSupporCuda()) {
            echo "Yes<br/>";
        } else {
            echo "No<br/>";
        }
    }
}

File: Java/Memory.java
class Memory extends Hardware {
    private int frequency;
    private int memorySize;
    private boolean supportsCuda;

    public Memory() {
    }

    public Memory(int frequency, int memorySize, boolean supportsCuda) {
        this.frequency = frequency;
        this.memorySize = memorySize;
        this.supportsCuda = supportsCuda;
    }

    public int getFrequency() {
        return frequency;
    }

    public void setFrequency(int frequency) {
        this.frequency = frequency;
    }

    public int getMemorySize() {
        return memorySize;
    }

    public void setMemorySize(int memorySize) {
        this.memorySize = memorySize;
    }

    public boolean isSupportsCuda() {
        return supportsCuda;
    }

    public void setSupportsCuda(boolean supportsCuda) {
        this.supportsCuda = supportsCuda;
    }

    public void printMemory() {
        System.out.println("Frequency : " + getFrequency() + " Mhz");
        System.out.println("Memory Size : " + getMemorySize() + " GB");
        System.out.print("Support Cuda : ");
        if (isSupportsCuda()) {
            System.out.println("Yes");
        } else {
            System.out.println("No");
        }
    }
}